### PR TITLE
Allow user to configure annotations on infra

### PR DESCRIPTION
**Pull request by arkodg**
This PR allows the user to add annotations to the
managed Envoy service and well as the Envoy pods using the EnvoyProxy resource

* Fixes https://github.com/envoyproxy/gateway/issues/377

* Closes https://github.com/envoyproxy/gateway/issues/648

### CODE REVIEW

File: api/config/v1alpha1/envoyproxy_types.go
Comment (zirain) [inline api/config/v1alpha1/envoyproxy_types.go:92]: should mention what's the default value.

Comment (arkodg) [reply api/config/v1alpha1/envoyproxy_types.go:92]: its `nil`

Comment (arkodg) [reply api/config/v1alpha1/envoyproxy_types.go:92]: added a follow up commit to address this, ptal

**Review by zirain — APPROVED** [review]

**Comment by arkodg** [comment]
thanks @zirain , hoping to get another approval before merging this API change

**Review by Xunzhuo — APPROVED** [review]
/lgtm
/approve

**Pull request merged**
